auto_inertial_params/proposal.md

```diff
@@ -0,0 +1,159 @@
+
+```
+<model name="model_name">
+  <link/>
+</model>
+```
+
+## Motivation
+
+Currently, there are 2 major workflows used by the users to obtain the correct inertial parameters of their models:
+
+ * Using CAD software like [Fusion360](https://www.autodesk.in/products/fusion-360/overview?term=1-YEAR&tab=subscription) or [Solidworks](https://www.solidworks.com/). Many users design their robot models using such CAD software which provide plugins that automatically generate the URDF/SDF for their model. Such plugins handle the calculation of the inertial parameters. For example, Fusion360 provides the [Fusion2URDF](https://github.com/syuntoku14/fusion2urdf) plugin which automatically generates a URDF with all the inertial parameters.
+
+ * Another way is to use 3rd-party Mesh Processing Software like [Meshlab](https://www.meshlab.net/). Such softwares take the mesh file as an input and provide the inertial parameters as an output which can then be copied and pasted into the URDF/SDF file. This is also the method that was suggested in official [Classic Gazebo docs](https://classic.gazebosim.org/tutorials?cat=build_robot&tut=inertia).
+
+Both of these ways create a dependency on external software and might be complicated for beginners. Native support for this feature directly into libsdformat would facilitate the effortless generation of accurate simulations.
+
+## User Perspective
+
+To specify the `<inertial>` element of a `<link>` in SDFormat, the user needs to add the `<mass>`, `<pose>`, and `<inertia>` tags. Here `<mass>` is the mass of the link and `<pose>` is the pose of the Centre of Mass with respect to the link frame. The `<inertia>` tag, on the other hand, needs to enclose the following 6 tags:
+
+ * `<ixx>` (Moment of Inertia)
+
+ * `<iyy>` (Moment of Inertia)
+
+ * `<izz>` (Moment of Inertia)
+
+ * `<ixy>` (Product of Inertia)
+ 
+ * `<ixz>` (Product of Inertia)
+ 
+ * `<izy>` (Product of Inertia)
+
+This proposal suggests the addition of the following in `SDFormat`: 
+
+ 1. An `auto` parameter for the `<inertia>` tag that would tell `libsdformat` to calculate Inertia matrix values automatically for the respective link. 
+
+ 2. A `//link/collision/material_density` tag for specifying the density of the collision geometry. This density value would be used to calculate the inertial parameters of the respective collision geometry. Adding this as part of the `<collision>` tag would allow a user to simulate links with different material types for different collisions. By default, the value of density would be set equal to that of water which is 1000 kg/m^3.  
```

Comment:
A link may contain 0 or multiple collision tags. The proposal should specify how inertial properties are computed in cases when there is not exactly 1 collision. If there are no collisions and the `auto` attribute is set to true, we could return an error. If there are multiple collisions, we could aggregate them using the `gz::math::Inertial` class and its `+` operator (see examples from its [unit test](https://github.com/gazebosim/gz-math/blob/gz-math7/src/Inertial_TEST.cc#L372-L621))

Reply:
@scpeters Thank you for the feedback. The case to handle multiple collisions is already under development [here](https://github.com/jasmeet0915/sdformat/tree/jasmeet/multi_collision_inertia) and would be added in [this](https://github.com/gazebosim/sdformat/pull/1299) PR. I'll also update the PR to handle a case of 0 collisions by returning an error and would update the proposal accordingly.

Reply:
Updated the section to include more information about different scenarios (multiple or no collisions) in [8b902d8](https://github.com/gazebosim/sdf_tutorials/pull/92/commits/8b902d835468ab130a8d2a832dedfd444e2013b0)